ecc_enclave/enclave/enclave.cpp

```diff
@@ -234,32 +246,101 @@ int ecall_cc_invoke(const char* encoded_args,
     {
         LOG_DEBUG("enclave crypto material created");
     }
+    */
 
+    bool b;
+    fpc_ChaincodeRequestMessage cc_request_message = {};
     t_shim_ctx_t ctx;
+    int ret;
+    // estimate max response len (take into account other fields and b64 encoding)
+    uint32_t response_len = b64_chaincode_response_message_len_in / 4 * 3 - 1024;
+    uint8_t response[b64_chaincode_response_message_len_in / 4 * 3];
+    uint32_t response_len_out = 0;
+    std::string b64_response;
+
     ctx.u_shim_ctx = u_shim_ctx;
-    ctx.encoded_args = encoded_args;
+    ctx.encoded_args = b64_chaincode_request_message;
 
-    // call chaincode invoke logic: creates output and response
-    // output, response <- F(args, input)
-    int ret;
-    if (strlen(pk) == 0)
-    {
-        // clear input
-        ctx.json_args = ctx.encoded_args;
-        ret = invoke(response, response_len_in, response_len_out, &ctx);
-    }
-    else
     {
-        // encrypted input
-        ret = invoke_enc(pk, response, response_len_in, response_len_out, &ctx);
+        // TODO decode b64_chaincode_request_message as necessary, and marshal parameters
+        // This block assumes b64_chaincode_request_message is literally what it is
+        pb_istream_t istream;
+
+        // base64 decode message
+        std::string decoded_crm;
+        std::string b64_chaincode_request_message_str(
+            b64_chaincode_request_message, b64_chaincode_request_message_len);
+        decoded_crm = base64_decode(b64_chaincode_request_message_str);
+
+        // set stream
+        istream =
+            pb_istream_from_buffer((const unsigned char*)decoded_crm.c_str(), decoded_crm.length());
+
+        b = pb_decode(&istream, fpc_ChaincodeRequestMessage_fields, &cc_request_message);
+        COND2LOGERR(!b, PB_GET_ERROR(&istream));
+        COND2LOGERR(cc_request_message.encrypted_request->size == 0, "zero size request");
+
+        // cc_request_message.encrypted_request->size);
+        ctx.encoded_args = (const char*)cc_request_message.encrypted_request->bytes;
```

Comment:
```suggestion
        // TODO: decrypt request
        // TODO: decrypted request should be CleartextRequestMessage rather than json args
        ctx.encoded_args = (const char*)cc_request_message.encrypted_request->bytes;
```

Reply:
done